File: app/src/main/java/com/backyardev/spacejam/fragments/HomeFragment.java
package com.backyardev.spacejam.fragments;

import android.os.Bundle;

import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.backyardev.spacejam.R;
import com.google.android.material.card.MaterialCardView;
import com.google.firebase.auth.FirebaseAuth;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

public class HomeFragment extends Fragment implements SwipeRefreshLayout.OnRefreshListener {
    private static final String TAG = "HomeFragment";


    private FirebaseAuth mAuth;
    private FirebaseAuth.AuthStateListener mAuthListener;

    private MaterialCardView cardView;
    private SwipeRefreshLayout homeRefreshLayout;


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate( R.layout.fragment_home, container, false );
        homeRefreshLayout = view.findViewById( R.id.homeSwipeRefresh );
        homeRefreshLayout.setOnRefreshListener( this );
        return view;
    }

    @Override
    public void onRefresh() {
        new MyTask().execute();
    }

    private class MyTask {

        void execute() {
            new Handler().postDelayed( new Runnable() {
                @Override
                public void run() {
                    if (homeRefreshLayout.isRefreshing()) {
                        homeRefreshLayout.setRefreshing( false );
                    }
                }
            }, 2000 );
        }
    }
}

File: app/src/main/java/com/backyardev/spacejam/fragments/HomeFragment.kt
package com.backyardev.spacejam.fragments

import android.os.Bundle

import android.os.Handler
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.backyardev.spacejam.R
import com.google.android.material.card.MaterialCardView
import com.google.firebase.auth.FirebaseAuth
import androidx.fragment.app.Fragment
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout

class HomeFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {


    private lateinit var mAuth: FirebaseAuth
    private lateinit var mAuthListener: FirebaseAuth.AuthStateListener

    private lateinit var cardView: MaterialCardView
    private lateinit var homeRefreshLayout: SwipeRefreshLayout


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_home, container, false)
        homeRefreshLayout = view.findViewById(R.id.homeSwipeRefresh)
        homeRefreshLayout!!.setOnRefreshListener(this)
        return view
    }

    override fun onRefresh() {
        MyTask().execute()
    }

    private inner class MyTask {

        internal fun execute() {
            Handler().postDelayed({
                if (homeRefreshLayout!!.isRefreshing) {
                    homeRefreshLayout!!.isRefreshing = false
                }
            }, 2000)
        }
    }

    companion object {
        private val TAG = "HomeFragment"
    }
}